### PR TITLE
Stanza filter

### PR DESCRIPTION
**Pull request by MarkGibbons**
### Description

Add the stanza filter.

### Issues Resolved

New feature.

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/line/145)

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: libraries/stanza_filter.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#
+
+# Filter insert or change keys in files formatted as stanzas
+module Line
+  class Filter
+    def stanza(current, args)
+      # Assumes stanzas are named uniquely across the file and contiguous
+      # Stanza starts with ^[<name>]$
+      # Stanza ends with next stanza or EOF
+      # Sets one instance of a key in a stanza to a new value TODO: set last found
```
Comment (JohnRoesler) [inline libraries/stanza_filter.rb]: Is this todo still valid? needs an issue?

Comment (MarkGibbons) [reply libraries/stanza_filter.rb]: Changing to WIP.  Can't get to it today.

Comment (JohnRoesler) [reply libraries/stanza_filter.rb]: No worries - just wanted to make sure it didn't get lost

Comment (MarkGibbons) [reply libraries/stanza_filter.rb]: This PR should be ready now.  Thanks for finding those TODOs.  I added tests to verify the behavior.

**Review by JohnRoesler — APPROVED** [review]

**Comment by MarkGibbons** [comment]
Thank you John.

**Comment by JohnRoesler** [comment]
You're welcome! I learned a lot about this cookbook and will probably use it in the future now 🙂

**Pull request merged**